pkgs/applications/editors/eclipse/default.nix

```diff
@@ -29,6 +29,18 @@ in rec {
             makeWrapper;
   };
 
+  ### Eclipse Platform
+
+  eclipse-platform = buildEclipse {
+    name = "eclipse-platform-${platform_major}.${platform_minor}";
+    description = "Eclipse Platform ${year}-${month}";
+    src =
+      fetchurl {
+        url = "https://www.eclipse.org/downloads/download.php?r=1&nf=1&file=/eclipse/downloads/drops${platform_major}/R-${platform_major}.${platform_minor}-${timestamp}/eclipse-platform-${platform_major}.${platform_minor}-linux-gtk-x86_64.tar.gz";
+        sha512 = "29hab3ha3spk0048k3mf2x5m80hlh1l6nazsykx0xxrqy9vdkdibv6mq74dzf1n93h1bd5qh9axicnhs465kp7r1irdl04761c1wibi";
+      };
+  };
+
```

Comment:
On reflection this move was unnecessary, but it's benign. The motivation I suppose is that this is the "most minimal" eclipse, so it goes first! It may also be possible to refactor the others to build on top of this one using the same mechanisms that are used to install plugins, which would simply things a bit ... but this is probably an "if it ain't broke don't fix it" situation.

Reply:
Note that this introduces a (trivial) conflict with https://github.com/NixOS/nixpkgs/pull/150757; you just need the new hash in this new location.